fpga_mem/fpga_datapath clarifications (#11387)
intel/llvm: sycl/doc/extensions/proposed/sycl_ext_intel_fpga_mem.asciidoc

allocation _T_, and a list of properties _PropListT_. The type of the
allocation _T_ also encodes the size of the allocation.

> `fpga_mem` initializes the underlying object `T` with the values passed into its constructor. If no values are passed into the constructor C++ default initialization rules will dictate the value of the underlying object `T`.

Author: ```suggestion
`fpga_mem` initializes the underlying object `T` with the values passed into its constructor. If no values are passed into the constructor the underlying object `T` is default initialized.
```

Below are some non-obvious interactions between `fpga_mem` and other SYCL extensions.

> * Since fpga_mem/fpga_datapath applies to the entire object that it wraps, conflicting hints may be applied the same object, such as in the example below.

Reviewer: As above
